Embedded System L1/AVR ATMEGA32/DIO/PROJECT/DIO_interface.h

```diff
@@ -0,0 +1,91 @@
+/******************************************************************************************************
+ * File: DIO_program.h 
+ * Author: Khalid Mosaad Zakaria
+ * Data: 10 july 2021
+ * Description: This File include Declarations of functions and Strucrures and some Macros.
+*******************************************************************************************************/
+
+#ifndef  DIO_interface_h_
+#define  DIO_interface_h_
+
+#define		PORTA	(u8)0
+#define		PORTB	(u8)1
+#define		PORTC	(u8)2
+#define		PORTD	(u8)3
+
+
+
+#define 	PIN0	(u8)0
+#define 	PIN1	(u8)1
+#define 	PIN2	(u8)2
+#define 	PIN3	(u8)3
+#define 	PIN4	(u8)4
+#define 	PIN5	(u8)5
+#define 	PIN6	(u8)6
+#define 	PIN7	(u8)7
+
+
+#define		INPUT	(u8)0
+#define		OUTPUT	(u8)1
+
+#define		HIGH	(u8)1
+#define		LOW		(u8)0
+
+
+typedef struct ST_DIO_config_t{
+u8 port_no;   // Port name
+u8 pin_no;	   // Pin name
+u8 state; // input or output
+}ST_DIO_config_t;
```

Comment:
it should be strDioConfig_t as the capitalization is needless and it is preferable to use str instead of ST
![image](https://user-images.githubusercontent.com/44152298/127505622-370594fc-ef66-4ee0-b7cd-120ce5afbf2a.png)
[Uploading Sprints_CodeGuideLine.pdf…]()